airflow/hooks/http_hook.py

```diff
@@ -108,13 +108,13 @@ def run(self, endpoint, data=None, headers=None, extra_options=None):
             url = (self.base_url or '') + (endpoint or '')
 
         req = None
-        if self.method == 'GET':
+        if self.method.upper() == 'GET':
             # GET uses params
             req = requests.Request(self.method,
                                    url,
                                    params=data,
                                    headers=headers)
-        elif self.method == 'HEAD':
+        elif self.method.upper() == 'HEAD':
```

Comment:
you won't have to do this if you do what I suggested in previous comment

Reply:
Updated PR as per your suggestion. Logic is now in constructor.